applications/ChimeraApplication/custom_utilities/hole_cutting_utility.h

```diff
@@ -0,0 +1,232 @@
+     * @param OverLapDistance is the the distance (magnitude) at which hole is to be cut from the zero distance layer.
+     * @param GetInside works in combination with MainDomainOrNot to get the feeling of what is inside or what is outside.
+     */
+    template<int TDim>
+    void RemoveOutOfDomainElements(ModelPart &rModelPart,
+                                   ModelPart &rModifiedModelPart,
+                                   const ChimeraHoleCuttingUtility::Domain DomainType,
+                                   const double OverLapDistance=0.0,
+                                   const ChimeraHoleCuttingUtility::SideToExtract Side = ChimeraHoleCuttingUtility::SideToExtract::OUTSIDE);
+
+    /**
+     * @brief Extracts the outside surface/edges of a modelpart.This uses the flag CHIMERA_INTERNAL_BOUNDARY
+     *                  to check if there is an internal boundary in the given ModelPart. The flag GetInternal
+     *                  specifies weather to get the internal boundary marked by CHIMERA_INTERNAL_BOUNDARY or the outside one.
+     * @param rVolumeModelPart The modelpart on which the boundary is to be found.
+     * @param rExtractedBoundaryModelPart The extracted surface/edge modelpart.
+     * @param GetInternal A bool specifying which surface/edge extracted. The one marked by CHIMERA_INTERNAL_BOUNDARY or the outside one.
+     */
+    template<int TDim>
+    void ExtractBoundaryMesh( ModelPart &rVolumeModelPart,
+                              ModelPart &rExtractedBoundaryModelPart,
+                              const ChimeraHoleCuttingUtility::SideToExtract GetInternal = ChimeraHoleCuttingUtility::SideToExtract::OUTSIDE);
+
+    /// Assignment operator.
+    ChimeraHoleCuttingUtility &operator=(ChimeraHoleCuttingUtility const &rOther) = delete;
+
+    /// Copy constructor.
+    ChimeraHoleCuttingUtility(ChimeraHoleCuttingUtility const& rOther) = delete;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    ///@}
```

Comment:
Remove.

Reply:
as mentioned this comes from the default structure of .h file in kratos. Any way. I removed. !

Reply:
True, but I think it makes not so much sense to keep them if empty... We should possible add something about this in the style guidelines.